x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/external/http/sender/HttpRequestSender.java

```diff
@@ -105,6 +107,14 @@ public void start() {
         }
     }
 
+    public Collection<RequestExecutorService.RateLimitingEndpointHandler> rateLimitingEndpointHandlers() {
+        if (service instanceof RequestExecutorService) {
```

Comment:
Instead of a getter let's have a new method that allows us to pass in the number of nodes that are being used to route requests, the new method will simply proxy the value on to the request executor service's new method. We can avoid the `instanceof` check here by adding the method in the interface `RequestExecutor`.

Reply:
Addressed with [Move rate limit division logic to RequestExecutorService](https://github.com/elastic/elasticsearch/pull/120400/commits/be9c5596808c0186df81298641be156e91add0d1)